solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrJdkClient.java

```diff
@@ -75,9 +77,16 @@ protected HttpSolrClientJdkImpl(String serverBaseUrl, HttpSolrClientJdkImpl.Buil
     if (builder.sslContext != null) {
       b.sslContext(builder.sslContext);
     }
+
     if (builder.executor != null) {
-      b.executor(builder.executor);
+      this.executor = builder.executor;
+      this.shutdownExecutor = false;
+    } else {
+      this.executor = Executors.newCachedThreadPool(Executors.defaultThreadFactory());
```

Comment:
I initially wanted to more-or-less mimic what `java.net.util.http.HttpClient` does if you do not pass in an Executor, and only reluctantly create my own because we need access to it for the pipes.  But deferring to `ForbiddenAPIs` I did switch it to the `ExecutorUtil` version (sorry I accidently pushed WIP here).  

I will mention here that I am a little concerned that `ExecutorUtil` has some things in it that really do not belong to the client, that exist only for inter-node requests.  For this reason I was hoping to avoid it.  See my other PR for one case where this could possibly cause trouble.  https://github.com/apache/solr/pull/2242

Reply:
It would be good to have a clearer distinction between the solrj classes that are meant for end user consumption vs the ones that are only ever to be used by the server.